packages/wrangler/src/api/form_data.ts

```diff
@@ -95,7 +96,11 @@ export function toFormData(worker: CfWorkerInit): FormData {
   );
 
   Object.entries(bindings.vars || {})?.forEach(([key, value]) => {
-    metadataBindings.push({ name: key, type: "plain_text", text: value });
+    if (typeof value === "string") {
+      metadataBindings.push({ name: key, type: "plain_text", text: value });
+    } else {
+      metadataBindings.push({ name: key, type: "json", json: value });
+    }
```

Comment:
I had a long think about this, because it looked kinda dangerous to just rely upon the `typeof` here. But I came to the conclusion that it is reasonable.

I am slightly concerned about the upstream consumption of these "JSON" values. We simply `JSON.stringify()` the whole `metadata` object. So I think we would end up sending a string to the server that is like:

**wrangler.toml**
```
[vars]
text = "plain ol' string"
count = 1
complex = { enabled = true, id = 123 }
```

**upload metadata string**
```
'{"main_module":"index.js",' + 
'"bindings":[{"name":"text","type":"plain_text","text":"plain ol' string"},' +
`{"name":"count","type":"json","json":1},' +
`{"name":"complex","type":"json","json":{"enabled":true,"id":123}}],' +
'"compatibility_date":"2022-01-12"}'
```

Note that the actual text that gets uploaded contains:

```
{"name":"complex","type":"json","json":{"enabled":true,"id":123}}
```

I would have expected the value of the `complex` var to be double encoded in JSON. so it would look like:

```
{"name":"complex","type":"json","json":"{\\"enabled\\":true,\\"id\\":123}"}
```

But perhaps I am over thinking this??